Graceful handling of broken lists when cuddled-lists extra is enabled

edufresne opened the pull request:
Fixes: https://github.com/trentm/python-markdown2/issues/577

## Overview
When `"cuddled-lists"` extra is passed, if you encounter ASCII art or any invalid list such as: `- - - Test`, an `AssertionError` is thrown due to the inability to parse it as a list and therefore the result text not containing the expected list structure.

## Changes
- Instead of throwing an assertion error, we simply treat the unparsed text as the paragraph text.
- Added a regression test to reproduce the assertion error

edufresne commented:
@nicholasserra anything needed on my end? Not sure why the builds above are failing^

nicholasserra commented:
Thanks! Probably a dependency issue with github actions, seems they removed 3.6 or something like that. Will check into it asap

nicholasserra commented:
Sorry for the delay. Just fixed CI, if you do a upstream rebase this should be OK now.

edufresne commented:
> Sorry for the delay. Just fixed CI, if you do a upstream rebase this should be OK now.

Awesome, thanks. Rebased. Once the workflow run is approved, should be good for a look if the build passes.

nicholasserra commented:
LGTM thank you!
